specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,438 @@
+        });
+
+        auto remoteId = L"{5AB838FB-B64B-4FCD-B25F-B09781D56DD5}";
+        auto channelOperation = PushNotificationChannelManager::CreatePushNotificationChannelAsync(remoteId);
+
+        // Setup the inprogress event handler
+        channelOperation.Progress(
+            [](
+                IAsyncOperationWithProgress<PushNotificationChannelResult, PushNotificationChannelResult> const& /* sender */,
+                PushNotificationChannelResult const& args)
+            {
+                if (args.Status() == ChannelStatus::InProgress)
+                {
+                    // This is basically a noop since it isn't really an error state
+                    printf("The first channel request is still in progress! \n");
+                }
+                else if (args.Status() == ChannelStatus::InProgressRetry)
+                {
+                    LOG_HR_MSG(args.ExtendedError(), "The channel request is in back-off retry mode because of a retryable error! Expect delays in acquiring it.");
+                }
+            });
+
+        // Setup the completed event handler
+        channelOperation.Completed(
+            [](
+                IAsyncOperationWithProgress<PushNotificationChannelResult, PushNotificationChannelResult> const& sender,
+                AsyncStatus const /* asyncStatus */)
+            {
+                auto result = sender.GetResults();
+                if (result.Status() == ChannelStatus::CompletedSuccess)
+                {
+                    auto channelUri = result.Channel().Uri();
+                    auto channelExpiry = result.Channel().ExpirationTime();
+
+                    // Persist the channelUri and Expiry in the App Service
+                }
+                else if (result.Status() == ChannelStatus::CompletedFailure)
+                {
+                    LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with channel request!");
+                }
+            });
+
+        // Draw window and other foreground UI stuff here
+
+        // Unsubscribe the foreground event
+        details.PushActivated(eventToken);
+    }
+
+    return 0;
+}
+```
+
+## Unregistering Push Activations
+In some certain rare cases, we may want to disable the activator for Push. For example, the user disables a toggle exposed by the app. To accomplish that, we would need to call into the UnRegister API
+```cpp
+int main()
+{
+    // Always Register the Push trigger as the first step: It sets up the Inproc COM server for subsequent Push operations
+    std::wstring activatorGuid = L"{B8E58D9D-A229-44A8-B923-2823C3791BCE}";
+    PushNotificationActivatorDetails details(activatorGuid);
+    PushNotificationChannelManager::RegisterPushNotificationActivator(details);
+
+    // Some code resides here
+
+    // This unregisters the Activator above
+    if (removePush)
+    {
+        PushNotificationChannelManager::UnregisterPushNotificationActivator(details.Registration());
```

Comment:
Why we need to pass `details.Registration()`? Shouldn't `details `be sufficient and let the Reunion layer to call Registration()? Just curious 😊.

Reply:
Excellent point!. Details should be sufficient. I will make that chnage.